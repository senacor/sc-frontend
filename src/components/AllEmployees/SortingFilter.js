import React from 'react';
import { injectIntl } from 'react-intl';
import { withStyles } from '@material-ui/core';

// Material UI
import FormControl from '@material-ui/core/FormControl';
import InputLabel from '@material-ui/core/InputLabel';
import Select from '@material-ui/core/Select';
import Input from '@material-ui/core/Input';
import MenuItem from '@material-ui/core/MenuItem';
import ListItemText from '@material-ui/core/ListItemText';
import Checkbox from '@material-ui/core/Checkbox';

const styles = theme => ({
  ...theme,
  expansionFilter: {
    display: 'block'
  },
  inputForAll: {
    width: 130,
    margin: 2 * theme.spacing.unit
  },
  inputForFormer: {
    width: 87,
    margin: 2 * theme.spacing.unit
  },
  chips: {
    display: 'flex',
    flexWrap: 'wrap'
  }
});

const SortingFilter = ({
  classes,
  sortBy,
  handleChange,
  menuData,
  stateValue,
  formerEmployees
}) => {
  return (
    <div className={classes.expansionFilter}>
      <FormControl className={classes.formControl}>
        <InputLabel htmlFor="select-multiple-positions">{sortBy}</InputLabel>
        <Select
          multiple
          value={stateValue}
          onChange={handleChange}
          input={
            <Input
              id="select-multiple-positions"
              className={
                formerEmployees ? classes.inputForFormer : classes.inputForAll
              }
            />
          }
          renderValue={selected => (
            <div className={classes.chips}>{`${selected.length} items`}</div>
          )}
        >
          {menuData.map(name => (
            <MenuItem key={name} value={name}>
              <Checkbox checked={stateValue.indexOf(name) > -1} />
              <ListItemText primary={name} />
            </MenuItem>
          ))}
        </Select>
      </FormControl>
    </div>
  );
};

export default injectIntl(withStyles(styles)(SortingFilter));
