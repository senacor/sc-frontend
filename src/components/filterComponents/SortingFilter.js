import React from 'react';
import { injectIntl } from 'react-intl';
import { withStyles } from '@material-ui/core';
// Material UI
import FormControl from '@material-ui/core/FormControl';
import InputLabel from '@material-ui/core/InputLabel';
import Select from '@material-ui/core/Select';
import Input from '@material-ui/core/Input';
import MenuItem from '@material-ui/core/MenuItem';
import ListItemText from '@material-ui/core/ListItemText';
import Checkbox from '@material-ui/core/Checkbox';

const styles = theme => ({
  ...theme,
  dropdownFilter: {
    display: 'block',
    marginBottom: 2 * theme.spacing.unit
  },
  input: {
    width: 120
  },
  chips: {
    display: 'flex',
    flexWrap: 'wrap'
  }
});

const SortingFilter = ({
  classes,
  sortBy,
  handleChange,
  menuData,
  stateValue,
  intl
}) => {
  const selectedTitle = countOfSelected => {
    return intl.formatMessage({
      id: countOfSelected < 2 ? 'filter.selected.item' : 'filter.selected.items'
    });
  };
  return (
    <div className={classes.dropdownFilter}>
      <FormControl className={classes.formControl}>
        <InputLabel htmlFor="select-multiple-positions">
          {intl.formatMessage({ id: sortBy })}
        </InputLabel>
        <Select
          multiple
          value={stateValue}
          onChange={handleChange}
          input={
            <Input id="select-multiple-positions" className={classes.input} />
          }
          renderValue={selected => (
            <div className={classes.chips}>
              {selected.length} {selectedTitle(selected.length)}
            </div>
          )}
        >
          {menuData.map(value => {
            const showText = () => {
              return sortBy === 'employeeInfo.scStatus'
                ? intl.formatMessage({ id: value })
                : value;
            };
            return (
              <MenuItem key={value} value={value}>
                <Checkbox checked={stateValue.indexOf(value) > -1} />
                <ListItemText primary={showText()} />
              </MenuItem>
            );
          })}
        </Select>
      </FormControl>
    </div>
  );
};

export default injectIntl(withStyles(styles)(SortingFilter));
