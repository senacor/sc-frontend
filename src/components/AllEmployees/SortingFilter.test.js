import React from 'react';
import { shallowWithIntl } from 'enzyme-react-intl';

import SortingFilter from './SortingFilter';

const mockMenuData = ['input1', 'input2', 'input3'];
const mockStateValue = 'searching employee';
const mockOnChange = jest.fn();

describe('SortingFilter', () => {
  const wrapper = shallowWithIntl(
    <SortingFilter
      menuData={mockMenuData}
      stateValue={mockStateValue}
      handleChange={mockOnChange}
    />
  );
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('checks for props values', () => {
    expect(wrapper.props().menuData).toHaveLength(3);
    expect(wrapper.props().stateValue).toBe('searching employee');
  });
  it('should have handleChange prop defined', () => {
    expect(wrapper.props().handleChange).toBeDefined();
  });
});
