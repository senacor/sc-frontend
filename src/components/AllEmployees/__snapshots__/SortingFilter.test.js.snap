// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortingFilter should render correctly 1`] = `
<WithStyles(SortingFilter)
  handleChange={[Function]}
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  menuData={
    Array [
      "input1",
      "input2",
      "input3",
    ]
  }
  stateValue="searching employee"
/>
`;
